internal/broker/labeling.go

```diff
@@ -0,0 +1,60 @@
+package broker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/k8s"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Labeler struct {
+	kcpClient client.Client
+	log       logrus.FieldLogger
+}
+
+func NewLabeler(kcpClient client.Client) *Labeler {
+	return &Labeler{
+		kcpClient: kcpClient,
+		log:       logrus.New(),
+	}
+}
+
+func (l *Labeler) UpdateLabels(id, newGlobalAccountId string) error {
+	kymaErr := l.updateCrLabel(id, k8s.KymaCr, newGlobalAccountId)
+	gardenerClusterErr := l.updateCrLabel(id, k8s.GardenerClusterCr, newGlobalAccountId)
+	runtimeErr := l.updateCrLabel(id, k8s.RuntimeCr, newGlobalAccountId)
+	err := errors.Join(kymaErr, gardenerClusterErr, runtimeErr)
+	return err
+}
+
+func (l *Labeler) updateCrLabel(id, crName, newGlobalAccountId string) error {
+	l.log.Infof("update labels starting for runtime %s for %s cr with new value %s", id, crName, newGlobalAccountId)
+	gvk, err := k8s.GvkByName(crName)
+	if err != nil {
+		return fmt.Errorf("while getting gvk for name: %s: %s", crName, err.Error())
+	}
+
+	var k8sObject unstructured.Unstructured
+	k8sObject.SetGroupVersionKind(gvk)
+	err = l.kcpClient.Get(context.Background(), types.NamespacedName{Namespace: KcpNamespace, Name: id}, &k8sObject)
+	if err != nil {
+		return fmt.Errorf("while getting k8s object of type %s from kcp cluster for instance %s, due to: %s", crName, id, err.Error())
+	}
+
+	err = k8s.AddOrOverrideMetadata(&k8sObject, k8s.GlobalAccountIdLabel, newGlobalAccountId)
```

Comment:
This name is misleading - you add or overwrite label only - not entire metatdata.

Reply:
I wonder if we can move this function to Labeler from k8s package.